website/docs/indexes.md

```diff
@@ -0,0 +1,126 @@
+---
+sidebar_position: 7
+---
+
+# Indexes
+
+Indexes are essential in improving query performance by enabling fast retrieval of relevant documents when querying large collections.
+
+Indexes in FerretDB are created based on a specific field(s) within documents.
+When an index is created, this creates a data structure that maps between the index values and the document locations, making it possible to retrieve documents more quickly based on those fields.
+
+## How do indexes work?
+
+Say a `products` collection contains the following documents:
+
+```js
+{ _id: 1, name: "iPhone 12", category: "smartphone", price: 799 }
+{ _id: 2, name: "iPad Pro", category: "tablet", price: 999 }
+{ _id: 3, name: "Galaxy S21", category: "smartphone", price: 699 }
+{ _id: 4, name: "MacBook Pro", category: "laptop", price: 1299 }
+```
+
+Suppose you want to find all products with a price greater than `$900`.
+Without an index, each query will need to scan each document in the collection to find the ones that match, which can be resource intensive.
+
+However, if there’s an index on the `price` field, the query only scans two documents where their prices are greater than $900, instead of all four.
+
+## How to create indexes
+
+Use the create use the createIndexes() command to create an index on a collection.
+This command takes two arguments: a document containing the index key (fields to index), and an optional document specifying the index options.
+
+You can create single field indexes or compound indexes.
+
+### Single Field Indexes
+
+Here's an example of creating an index on the `price` field of a products collection:
+
+```js
+db.products.createIndex({ price: 1 })
+```
+
+This creates an ascending index on the `price` field.
+`1` specifies the index direction for ascending order.
+If it’s `-1`, it specifies a descending order direction for the index.
+
+### Compound Indexes
+
+For compound indexes, you can create an index key combining multiple fields together as a key.
+Below is an example of a compound index that uses `price` and `category` fields from the `products` collection as the index key.
+
+```js
+db.products.createIndex({ price: 1, category: 1 })
+```
+
+:::note
+
+* If the createIndex() command is called for a non-existent collection, it will create the collection and its given indexes.
+* If the createIndex() command is called for a non-existent field, only an index for the field is created.
```

Comment:
This sounds a bit strange. Does this mean that it creates an index but does not create a field right?